samples/extensions/full_screen_exclusive/full_screen_exclusive.cpp

```diff
@@ -0,0 +1,1172 @@
+	initialize_windows();        // this can be only called after GetActiveWindow(), or as long as HWND_application_window represent the correct handle!
+
+	if (!validate_extensions(active_instance_extensions, instance_extensions))
+	{
+		throw std::runtime_error("Required instance extensions are missing.");
+	}
+	uint32_t instance_layer_count;
+	VK_CHECK(vkEnumerateInstanceLayerProperties(&instance_layer_count, nullptr));
+	std::vector<VkLayerProperties> supported_validation_layers(instance_layer_count);
+	VK_CHECK(vkEnumerateInstanceLayerProperties(&instance_layer_count, supported_validation_layers.data()));
+	std::vector<const char *> requested_validation_layers(required_validation_layers);
+
+#ifdef VKB_VALIDATION_LAYERS
+	// Determine the optimal validation layers to enable that are necessary for useful debugging
+	std::vector<const char *> optimal_validation_layers = vkb::get_optimal_validation_layers(supported_validation_layers);
+	requested_validation_layers.insert(requested_validation_layers.end(), optimal_validation_layers.begin(), optimal_validation_layers.end());
+#endif
+
+	if (validate_layers(requested_validation_layers, supported_validation_layers))
+	{
+		LOGI("Enabled Validation Layers:")
+		for (const auto &layer : requested_validation_layers)
+		{
+			LOGI(" \t{}", layer)
+		}
+	}
+	else
+	{
+		throw std::runtime_error("Required validation layers are missing.");
+	}
+
+	VkApplicationInfo app{VK_STRUCTURE_TYPE_APPLICATION_INFO};
+	app.pApplicationName = "Full Screen Exclusive";
+	app.pEngineName      = "Vulkan Samples";
+	app.apiVersion       = VK_MAKE_VERSION(1, 0, 0);
+
+	VkInstanceCreateInfo instance_info{VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO};
+	instance_info.pApplicationInfo        = &app;
+	instance_info.enabledExtensionCount   = vkb::to_u32(active_instance_extensions.size());
+	instance_info.ppEnabledExtensionNames = active_instance_extensions.data();
+	instance_info.enabledLayerCount       = vkb::to_u32(requested_validation_layers.size());
+	instance_info.ppEnabledLayerNames     = requested_validation_layers.data();
+
+#if defined(VKB_DEBUG) || defined(VKB_VALIDATION_LAYERS)
+	VkDebugReportCallbackCreateInfoEXT debug_report_create_info = {VK_STRUCTURE_TYPE_DEBUG_REPORT_CREATE_INFO_EXT};
+	debug_report_create_info.flags                              = VK_DEBUG_REPORT_ERROR_BIT_EXT | VK_DEBUG_REPORT_WARNING_BIT_EXT;
+	debug_report_create_info.pfnCallback                        = debug_callback;
+
+	instance_info.pNext = &debug_report_create_info;
+#endif
+
+#if (defined(VKB_ENABLE_PORTABILITY))
+	instance_info.flags |= VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
+#endif
+
+	VK_CHECK(vkCreateInstance(&instance_info, nullptr, &context.instance));
+	volkLoadInstance(context.instance);
+
+#if defined(VKB_DEBUG) || defined(VKB_VALIDATION_LAYERS)
+	VK_CHECK(vkCreateDebugReportCallbackEXT(context.instance, &debug_report_create_info, nullptr, &context.debug_callback));
```

Comment:
This crashes for me on a secondary Workstation that doesn't have all the developement stuff setup. `vkCreateDebugReportCallbackEXT` is zero, and I think the problem might be related to how the preprocessor define are checked.

Validation layers (line 175) only check for `VKB_VALIDATION_LAYERS`, but the callback creation also has on or clause with `VKB_DEBUG`:

```cpp
#ifdef VKB_VALIDATION_LAYERS
	// Determine the optimal validation layers to enable that are necessary for useful debugging
	std::vector<const char *> optimal_validation_layers = vkb::get_optimal_validation_layers(supported_validation_layers);
	requested_validation_layers.insert(requested_validation_layers.end(), optimal_validation_layers.begin(), optimal_validation_layers.end());
#endif
...
#if defined(VKB_DEBUG) || defined(VKB_VALIDATION_LAYERS)
	VK_CHECK(vkCreateDebugReportCallbackEXT(context.instance, &debug_report_create_info, nullptr, &context.debug_callback));
#endif
```

On this system, where I did a clean checkout and did not change any settings this results in the aforementioned crash. Haven't been able to test on my personal desktop system yet.

All preprocessor defines should check for the same condition.

Reply:
great catch.  All defines now agree.